### PR TITLE
Support directly designating a gitSSHKey instead of File for launcher

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does / why we need it**:

Add `--git-ssh-key-data` for launcher.  This flag enables passing a sshKey without preparing a file.

Usecase:
- You can use the same launcher command for AWS and GCP (+ etc.)
- If you use AWS Secrets Manager to store the sshKey, Secrets Manager will write it to an env (e.g. `YOUR_ENV`) and you can read it by `/bin/sh -c \"launcher launcher (...) --git-ssh-key-data=$YOUR_ENV \"` 
    - cf. https://pipecd.dev/docs-v0.49.x/installation/install-piped/installing-on-fargate/#installation

**Which issue(s) this PR fixes**:

N/A

**Does this PR introduce a user-facing change?**: N/A
 
- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Review by Warashi — APPROVED** [review]
👍🏻

**Comment by t-kikuc** [comment]
@Warashi 
Sorry, would you please review it again?

I changed from `--git-ssh-key-env` to `--git-ssh-key-data` since it's easier to use.

File: pkg/app/launcher/cmd/launcher/launcher.go [hunk 1]
```diff
@@ -227,6 +233,23 @@ func (l *launcher) run(ctx context.Context, input cli.Input) error {
...
+		if l.gitSSHKeyData != "" {
+			decodedKey, err := base64.StdEncoding.DecodeString(l.gitSSHKeyData)
+			if err != nil {
+				return fmt.Errorf("failed to decode SSH key data, (%w)", err)
+			}
+			tmpKeyFile, err := os.CreateTemp("", "git-ssh-key-data")
+			if err != nil {
+				return fmt.Errorf("failed to create a temp file for SSH key data (%w)", err)
+			}
+			if _, err = tmpKeyFile.WriteString(string(decodedKey)); err != nil {
```
Comment (Warashi) [inline pkg/app/launcher/cmd/launcher/launcher.go:245]: nits.
We can use `Write` method as belows

```suggestion
			if _, err = tmpKeyFile.Write(decodedKey); err != nil {
```

**Review by Warashi — APPROVED** [review]
Thanks

I commented on a nitpick, but you can merge this, leaving it as-is.

File: pkg/app/launcher/cmd/launcher/launcher.go (hunk 1 above)
Comment (t-kikuc) [reply pkg/app/launcher/cmd/launcher/launcher.go]: @Warashi 

thank you, I fixed it here https://github.com/pipe-cd/pipecd/pull/5258/commits/1c6e430f6ae1c3c35a72f723b7d262198ca0775b.

It's a legacy code and I missed.

**Review by Warashi — APPROVED** [review]
Thanks

**Review by khanhtc1202 — APPROVED** [review]
LGTM :+

**Pull request merged**